geyser-plugin-manager/src/entry_notifier.rs

```diff
@@ -0,0 +1,76 @@
+/// Module responsible for notifying plugins about entries
+use {
+    crate::geyser_plugin_manager::GeyserPluginManager,
+    log::*,
+    solana_entry::entry::Entry,
+    solana_geyser_plugin_interface::geyser_plugin_interface::{
+        ReplicaEntryInfo, ReplicaEntryInfoVersions,
+    },
+    solana_measure::measure::Measure,
+    solana_metrics::*,
+    solana_rpc::entry_notifier_interface::EntryNotifier,
+    solana_sdk::clock::Slot,
+    std::sync::{Arc, RwLock},
+};
+
+pub(crate) struct EntryNotifierImpl {
+    plugin_manager: Arc<RwLock<GeyserPluginManager>>,
+}
+
+impl EntryNotifier for EntryNotifierImpl {
+    fn notify_entry<'a>(&'a self, slot: Slot, index: usize, entry: &'a Entry) {
+        let mut measure = Measure::start("geyser-plugin-notify_plugins_of_entry_info");
+
+        let plugin_manager = self.plugin_manager.read().unwrap();
+        if plugin_manager.plugins.is_empty() {
+            return;
+        }
+
+        let entry_info = Self::build_replica_entry_info(slot, index, entry);
```

Comment:
While a lot of this file is copied from transaction_notifier.rs, I moved this line down, as there's no reason to build the entry info if `plugins.is_empty()`. This is probably inconsequential in terms of perf, but we could make this change in the other notifiers.